numpy/core/src/multiarray/nditer_pywrap.c

```diff
@@ -927,7 +927,7 @@ NpyIter_NestedIters(PyObject *NPY_UNUSED(self),
         }
         for (i = 0; i < nested_naxes[inest]; ++i) {
             PyObject *v = PySequence_GetItem(item, i);
-            npy_intp axis;
+            int axis;
```

Comment:
Hmm, changing the nditer code seems risky. Is there any reason it should *not* be npy_intp?  A bit of extra precision is not harmful.